launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java

```diff
@@ -336,4 +334,18 @@ static void addPermGenSizeOpt(List<String> cmd) {
     cmd.add("-XX:MaxPermSize=256m");
   }
 
+  /**
+   * Get the major version of the java.version string supplied.
+   */
+  static int javaMajorVersion(String javaVersion) {
+    String[] version = javaVersion.split("[+.\\-]+");
```

Comment:
It probably isn't much harm; the only concern would be that it split string, the test and the comment together imply a different input format than the one specified. It might be slightly nicer to have it match the spec since its just a few characters difference. (but a super minor point)